### ISSUE TITLE
Gamepad support

### ISSUE THREAD

**Issue by Sp4iK** [matzman666/OpenVR-AdvancedSettings#13]
Hello,

I'm trying to use this with my Xbox 360 gamepad and it is not responding to any button. Is this a bug or only Vive controls are supported?

**Comment by matzman666** [comment]
Currently gamepads are not supported.  OpenVR does not provide any interface that allows me to support gamepads ([Related issue](https://github.com/ValveSoftware/openvr/issues/284)). 

The only way would be to use steamworks for the steam controller (which I cannot do because of license issues) and XInput for all other controllers (which I cannot do because I only have a steam controller, and I cannot develop something which I cannot test).

As long as Valve does not provide a controller api under a permissive license I cannot support controllers.

**Comment by Balderick** [comment]
current steam client beta married with steamvr beta gives 
<img width="956" alt="xbox steamvr" src="https://cloud.githubusercontent.com/assets/6255884/21747227/dc82f31a-d557-11e6-96ac-2900f95ff20c.PNG">

Have Valve got passed those licensing issues?

Could xpadder or freepie be used to map a rudimentary set of vive controller actions (or keyboard equivalents) to allow xbox controller to work with advanced settings?

**Comment by matzman666** [comment]
No, I am not aware of any license changes (I am sure it would be all over the internet when they do change license).

No, any third-party program would be having a hard time injecting vive controller events into any application (yet). These third-party programs use the windows API for injecting events, and motion controllers are not managed by any windows API yet, they would need to hack OpenVR.

**Comment by Balderick** [comment]
ahh so the windows creator update could bring openvr implemntation.  interesting.

**Comment by gormanj** [comment]
What about mouse control? There seems to be no way to navigate with no motion controller.

**Comment by matzman666** [comment]
> What about mouse control? There seems to be no way to navigate with no motion controller.

Advanced Settings is a background application. It has no window that is eligible for input focus and therefore cannot receive mouse events.

**Comment by Kbeam007** [comment]
It is actually possible to emulate Razer Hydra with FreePIE based on gamepad input, so everyone with a gamepad can mavigate the settings.

**Comment by Balderick** [comment]
True.  Its possible to do that but its clunky, very clunky.

Hopefully prices for steamvr tracking accessories like steamvr base stations, steamvr device dongles and steamvr controllers will start to come down as more and more manufacturers release steamvr tracking based products to consumers.

Other so!utions like nolovr tracking system brings room scale vr to the masses at an affordable price.

**Comment by Kbeam007** [comment]
The Hydra emulation is suitable for experiences, some basic input and for simple puzzles, but not for other controls that require fast and precise operation. It still enabled me to play games like Mind Path to Thalamus and now to access OpenVR Advanced Settings. 

PSMoveService is much better, of course, but you still get the bare minimum with FreePIE.

**Comment by Balderick** [comment]
Would making advanced-settings available through steam store or steamvr workshop not help with making this app more steam friendly.?

We need advanced-settings compiled with steam runtime somehow, any ideas?

**Comment by Balderick** [comment]
> It is actually possible to emulate Razer Hydra with FreePIE based on gamepad input, so everyone with a gamepad can mavigate the settings.

@Kbeam007 could you share your freepie script? or describe your working setup?

**Comment by Balderick** [comment]
> The only way would be to use steamworks for the steam controller (which I cannot do because of license issues) and XInput for all other controllers (which I cannot do because I only have a steam controller, and I cannot develop something which I cannot test).

@matzman666 Maybe steamvr needs an application category in steam workshop?  Or advance settings downloadable through steam store or steam tools?  

I really agree this is an openvr issue as any app listed in steamvr dashboard should be able to use any input device that steamvr apps support.  I can use the buttons on my wheel base for steamvr dashboard, steamvr apps, steam overlays but not in openvr advanced settings even though steamvr lists it as an application.

What is the best (lightest) way you know of getting a steam controller or Xbox controller to work with advancedsettings?.

**Comment by Kbeam007** [comment]
@Balderick The FreePIE forum has a thread on that with instructions and .dll files that should substitute the nornam Hydra drivers http://www.mtbs3d.com/phpBB/viewtopic.php?f=139&t=22033&start=40 . My experience with the .dll files is that for some reason the pitch and roll values are with regard to the world axes and the yaw values are with regard to the local space of the controller. I tried a very crude "fix" in my script here (slightly outdated, but does the job in most situations) https://drive.google.com/file/d/0B_7L0fiUQpSYOTA5aEJ6LV9Pa0k/view?usp=sharing

**Comment by Balderick** [comment]
I found the DLLs that come with freepie were  not doing their job but the ones bundled with the files shared with https://youtu.be/GQ8up6uS5zM do work.

IMO we simply should not be having to emulate vive controllers if no vive controllers are available.  Like anything else there will be a reason that we do but I can not see it as steamvr, steamvr apps, big picture and in game overlays all support Xbox controller.    

Instead of emulating positionally tracked controllers I reckon we need to exactly replicate how vive controllers are configured for steam overlays.   We really just need the menu navigation inputs.  

Could anybody share how vive controller is configured for steamvr overlays by sharing the contents of the json being used specifically for overlays I.e. dashboard, in game overlay etc?  

Do the vive controllers show up as xinput devices listed as non steam controllers in controller settings accessed through big picture without steamvr running?

**Comment by mungewell** [comment]
Hi, I'd second the request for gamepad/mouse control, or at least something that does not require hand controllers to be present (or emulated, PSMoves?).
I am currently experimenting with TrinusPSVR, and since racing Sims are my thing I have no real need to set up hand controllers.
Is there a way to have the render target be a 'normal' window, rather than on the in-VR dock?
